WordPress/gutenberg
# Template Editing Mode: Fix options dropdown

## Description
PR makes the following changes in the Template Title component:

* Remove template descriptions. Templates visible in the post/page dropdown currently don't support this feature.
* Conditionally hides dropdown when the template has no actionable options - change the title or delete.

P.S. We should change the "Delete" text to "Clear customizations" when a template has a theme file. I will address that in separate PR since we're in string freeze, and I want this to be backported in beta 4.

Resolves #37427.

## How has this been tested?
Using TT2

1. Create/edit a page.
2. Create a new template.
3. Make sure the "Template Options" dropdown is visible and works.
4. Switch to custom theme templates and see that only template title is displayed.

## Screenshots 
| with options | without options |
| --- | --- |
|![CleanShot 2021-12-16 at 12 10 34](https://user-images.githubusercontent.com/240569/146334469-b1a3e56f-4f48-4795-a250-6c145719dcbf.png)|![CleanShot 2021-12-16 at 12 10 52](https://user-images.githubusercontent.com/240569/146334472-6f2be32e-0ec0-4617-9a5d-6fec9bedce65.png)|


## Types of changes
Bugfix

## Checklist:
